transcom/mymove
# MAIN - B-21527 counselor actual expense reimbursement

## [B-21527](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-21527)

INT PR: https://github.com/transcom/mymove/pull/14039
Added tests: https://github.com/transcom/mymove/pull/14064

## Summary

This adds the frontend functionality for PPM field - "Actual Expense Reimbursement". This is a editable field on the Edit Shipments page for PPMs and when reviewing PPM closeout documents using the edit icon.

- Added 'isActualExpenseReimbursement' radio buttons to PPM edit shipment page and closeout review page - edit modal.
- When pay grade is changed to `Civilian Employee` -> `isActualExpenseReimbursement` is auto changed to `true` for all PPM shipments.
- Radio buttons are disabled for Civilian moves and value for `isActualExpenseReimbursement` should be `true`(yes).
- `Actual Expense Reimbursement` tags are added to Move details page(shipment card), add/edit PPM shipment, and PPM closeout document review page.


### How to test

1. As a customer -> while creating a move, choose `Civilian Employee` as Pay grade and create a PPM shipment.
2. Sign and submit the move.
3. As SC -> find the move and verify you see `ACTUAL EXPENSE REIMBURSEMENT` tag on the PPM shipment card in Move details page.
4. Click Edit shipment and verify raido buttons for `Is this PPM an Actual Expense Reimbursement?` is checked on `Yes` but disabled.
5. Go through the PPM steps to get to closeout review documents page or use `ApprovedMoveWithPPMProgearWeightTicketOfficeCivilian` testharness for ease.
6. As a SC -> search for the move and click "Review documents" in move details.
7. You should see the `ACTUAL EXPENSE REIMBURSEMENT` tag and Actual Expense Reimbursement should be `Yes` under Shipment Info and Edit icon should be disabled.
8. Repeat 1-7 with a move that is not a Civilian employee pay grade and verify that you can edit as SC and also `No` is checked.
9. After creating a PPM shipment move-> if you change pay grade to a Civilian -> `Actual Expense Reimbursement` should auto change to `true`(yes).

## Screenshots
![image](https://github.com/user-attachments/assets/484aaefe-1419-46fe-8c04-8684ef90a69b)
![image](https://github.com/user-attachments/assets/e86f8840-c468-41f5-98e3-1645e1fc3870)
![image](https://github.com/user-attachments/assets/1b7d5f9e-1360-4cc7-a018-c42f8b2677d0)
